README.md
damacreat_io
===========
[Play on kayzgames.github.io](http://kayzgames.github.io/damacreat_io/)


CHANGELOG.md
# 0.11.3 (2021-02-09)

## Bugfix
* game could be joined before connection to server was fully initialised

# 0.11.2 (2020-09-27)

## Bugfix
* fixed a bug in a shader that only happend on some systems

# 0.11.1 (2020-09-19)

## Bugfix
* pack shaders into the source to prevent errors because of caching issues (and reduce the amount of requests and improve loading times)

# 0.11.0 (2020-09-17)

## Misc
* improved performance
* animated and more colorful background
* better handling of clients that don't support webgl

## Bugfixes
* size of moving dark energy off screen was not updated correctly
* serverside bug that crashed the server after a several months

# 0.10.1 (2019-05-23)

## Bugfixes
* fixed extreme drop in performance on firefox when rendering a black hole

# 0.10.0 (2019-05-22)

## Gameplay
* increased loss of size over time for bigger damacreats and reduced for smaller damacreats
* generate less dark energy when a player disconnects
* moving dark energy slowly dissipates even if it's not being absorbed by a damacreat
* added black holes to the world, black holes reduce the size of nearby damacreats (as long as the damacreat has a score of at least 30)
* damacreats can fire black holes with their black hole cannon (can be fire before it's loaded fully, but the resulting black hole will be smaller and shorter lived)

## Misc
* improved analytics
* visibility of leaderboard can be toogled
* added keyboard shortcuts for nicknames, leaderboard, fps, debug info and the minimap

## Bugfixes
* fixed several bugs introduced by the new netcode
* exhaust dark energy created by boosting was not correctly displayed when created by very large damacreats

# 0.9.0 (2019-04-06)

## Gameplay
* dark energy gets digested faster
* reduced max normal speed of players by 25% 
* increased boosted speed of players by 500%
* increased amount of food by 50%
* added bots (maximum of 20 bots, no bots spawn if there are at least 40 damacreats)

## Misc
* made digestion of dark energy visually more obvious
* better default names
* improved netcode and reduced amount of traffic
* added link to feedback form
* added link to isowosi Twitter

## Bugfixes
* fixed a source of several (actual and potential) bugs

# 0.8.0 (2018-11-17)

## Gameplay
* less and randomly distributed dark energy when a player disconnects

## Misc
* added initial gamepad support; just press start on your gamepad and you are in
* improved visuals of normal movement
* improved visuals of boosted movement

# 0.7.0 (2018-11-07)

## Gameplay
* using the boost reduces the size of the damacreat and produces dark energy
* food below a certain size is consumed instantly
* a player that disconnects or has a radius smaller than 5 will be turned into several pieces of dark energy 

## Misc
* performance improvement when there is a lot of moving food/dark energy
* the player can pick their color

## Bugfixes
* QuadTree wasn't updated correctly for moving stuff between two nodes resulting in performance degration over time
* improved rendering quality

# 0.6.0 (2018-09-25)

## Gameplay
* the maximum velocity of a damacreat depends on its size
* damacreats can use a temporary speed boost
* modified digestion speed - faster for small ones and slower for big ones

## Misc
* performance improvement updating the quadtree

# 0.5.1 (2018-09-22)

## Gameplay
* added a minimap

## Misc
* use Roboto font instead of Verdana everywhere
* show version in menu
* added changelog
* remember nickname when losing
* start game on enter
* store settings in browser
* added Google Analytics, updated privacy policy

## Bugfixes
* background was not rendered on mobile phones since 0.4.0, switched back to highp precision and disabled background rendering on devices not supporting highp precision in fragment shaders

# 0.5.0 (2018-09-19)

## Gameplay
* damacreats can eat other damacreats
* zoom depends on the size of the player

## Misc
* display error if connection to server fails
* improved leaderboard

## Bugfixes
* fixed a bug causing unneccessary network traffic when a player disconnects while consuming dark energy
* pinch zoom on mobile was still possible

# 0.4.0 (2018-09-16)

## Misc
* show player names
* disabled zooming
* improved general layout on mobile

## Bugfixes
* increased speed limit of vomitted dark energy
* layout on mobile when entering a name got messed up
* game could not start on devices not supporting highp in fragment shaders

# 0.3.0 (2018-09-13)

## Misc
* added a menu
* added privacy policy and imprint
* show errors if there are are errors when loading the game
* added settings and hid debug data by default

# 0.2.0 (2018-09-10)

## Gameplay
* dark energy bounces off borders

## Misc
* added touch control
* improved coloring of damacreats and background
* improved visuals of dark energy
* more colors
* switched from webgl2 to webgl1

## Bugfixes
* fixed shaking of damacreats
* background wasn't renered completely in some resolutions
* removed scrollbars on mobile
* fixed various visual bugs related to running the game in the background

# 0.1.0 (2018-09-07)

* Initial version


main.dart.js_1.part.js
self.$__dart_deferred_initializers__=self.$__dart_deferred_initializers__||Object.create(null)
$__dart_deferred_initializers__.current=function(a,b,c,$){var C,H,J,P,W,G,Y,K,M,Q={
C3:function(d,e){var x,w=new Q.mb(E.ea(d,e,3)),v=$.yC
if(v==null)v=$.yC=O.dE($.F7,null)
w.b=v
x=document.createElement("privacy-policy")
w.c=x
return w},
Fv:function(d,e){return new Q.j8(E.br(d,e))},
mb:function mb(d){var _=this
_.c=_.b=_.a=_.f=_.e=null
_.d=d},
j8:function j8(d){var _=this
_.d=_.c=_.b=null
_.a=d}},D,O,V,E,A,R,T,L,N,U,X,Z,F,B,S
a.setFunctionNamesIfNecessary([Q])
C=c[0]
H=c[1]
J=c[2]
P=c[3]
W=c[4]
G=c[5]
Y=c[6]
K=c[7]
M=c[8]
Q=a.updateHolder(c[9],Q)
D=c[10]
O=c[11]
V=c[12]
E=c[13]
A=c[14]
R=c[15]
T=c[16]
L=c[17]
N=c[18]
U=c[19]
X=c[20]
Z=c[21]
F=c[22]
B=c[23]
S=c[24]
Q.mb.prototype={
R:function(){var x,w,v,u,t,s,r,q,p,o,n,m,l,k,j,i,h,g,f,e,d,a0,a1,a2,a3,a4=this,a5="h4",a6="p",a7=a4.a,a8=a4.bA(),a9=document,b0=T.aO(a9,a8)
a4.aa(b0,"exit")
a4.u(b0)
x=T.J(a9,b0,"button")
a4.u(x)
T.F(x,"Close")
w=T.J(a9,a8,"h2")
a4.L(w)
T.F(w,"Privacy Policy")
v=T.J(a9,a8,"h3")
a4.L(v)
T.F(v,"General")
u=T.J(a9,a8,a5)
a4.L(u)
T.F(u,"Data Processing by isowosi UG (haftungsbeschr\xe4nkt)")
t=T.J(a9,a8,a6)
a4.L(t)
T.F(t,"While using this website, only the technical data required to deliver the requested resources and play the game is processed.")
s=T.J(a9,t,"ul")
a4.u(s)
r=T.J(a9,s,"li")
a4.L(r)
T.F(r,"IP-address to deliver resources to the user")
q=T.J(a9,s,"li")
a4.L(q)
T.F(q,"data which is send by the browser by default (HTTP Request Header)")
p=T.J(a9,s,"li")
a4.L(p)
T.F(p,"inputs you provide to play the game (mouse movement, touch screen input, etc.)")
o=T.J(a9,a8,a5)
a4.L(o)
T.F(o,"Data Storage by isowosi UG (haftungsbeschr\xe4nkt)")
n=T.J(a9,a8,a6)
a4.L(n)
T.F(n,"No personally identifiable information (PII) is stored while using this website.")
m=T.J(a9,a8,a6)
a4.L(m)
T.F(m,"Game settings are stored using the browser storage.")
l=T.J(a9,a8,"h3")
a4.L(l)
T.F(l,"Data Processing and Data Storage by third parties")
k=T.J(a9,a8,a5)
a4.L(k)
T.F(k,"Hosting")
j=T.J(a9,a8,a6)
a4.L(j)
T.F(j,"To host this website, the infrastructure of GitHub is used. GitHub is certified under the US Privacy Shield and complies with the GDPR.")
i=T.J(a9,a8,a6)
a4.L(i)
T.F(i,"For the serverside game logic the game connects to the infrastructure of Digitial Ocean. Digitial Ocean is certified under the US Privacy Shield and complies with the GDPR.")
h=T.J(a9,a8,a5)
a4.L(h)
T.F(h,"Google Fonts")
g=T.J(a9,a8,a6)
a4.L(g)
T.F(g,"This website uses the font Roboto provided by Google Fonts. Google Fonts limits the processing and storing of personally identifiable information to the minimum needed to efficiently use fonts. No authentification is required and no cookies are stored. Google is certified under the US Privacy Shield and complies with the GDPR.")
f=T.J(a9,a8,a5)
a4.L(f)
T.F(f,"Google Analytics")
e=T.J(a9,a8,a6)
a4.L(e)
T.F(e,"This website uses Google Analytics to collect usage statistics. Google Analytics stores cookies in your browser to distinguish between first time visitors and returning visitors. IP anonymization is configured for Google Analytics. Google is certified under the US Privacy Shield and complies with the GDPR.")
d=T.J(a9,a8,a6)
a4.L(d)
T.F(d,'If you have "do not track" (DNT) enabled in your browser, Google Analytics is disabled on this website (and can not be enabled).')
a0=a4.e=new V.aM(42,a4,T.c0(a8))
a4.f=new K.ba(new D.b_(a0,Q.EU()),a0)
a1=T.aO(a9,a8)
a4.aa(a1,"exit")
a4.u(a1)
a2=T.J(a9,a1,"button")
a4.u(a2)
T.F(a2,"Close")
a0=a7.gfG()
a3=y.d
J.fq(x,"click",a4.b0(a0,a3))
J.fq(a2,"click",a4.b0(a0,a3))},
a3:function(){var x=this.a
this.f.saR(!x.gcz().r)
this.e.am()},
aE:function(){this.e.at()}}
Q.j8.prototype={
R:function(){var x,w,v=this,u=document,t=u.createElement("label")
v.L(t)
x=T.J(u,t,"input")
v.d=x
T.O(x,"type","checkbox")
v.u(v.d)
T.F(t," allow Google Analytics to track you on this website")
x=v.d
w=y.d;(x&&C.j).ak(x,"change",v.bf(v.gmf(),w,w))
v.ai(t)},
a3:function(){var x,w=this,v=w.a.a,u=v.gcz(),t=!u.r&&u.x
u=w.b
if(u!=t){w.d.checked=t
w.b=t}x=v.gcz().r
u=w.c
if(u!==x){w.d.disabled=x
w.c=x}},
mg:function(d){var x=this.a.a,w=x.gcz(),v=x.gcz()
w.sib(!(!v.r&&v.x))}}
var z=a.updateTypes(["~(@)","c5<~>*(W*,i*)"]);(function installTearOffs(){var x=a._static_2,w=a._instance_1u
x(Q,"EU","Fv",1)
w(Q.j8.prototype,"gmf","mg",0)})();(function inheritance(){var x=a.inherit
x(Q.mb,E.dF)
x(Q.j8,E.c5)})()
H.j3(b.typeUniverse,JSON.parse('{"jr":"r","kn":"r","jq":"u","kz":"u","nP":"aZ","jy":"y","l1":"y","n9":"C","mi":"bw","lq":"aE","md":"aL","jW":"bd","k5":"be","jO":"av","lP":"av","kE":"by","jY":"X","k0":"aK","mb":{"W":[],"a6":[]},"j8":{"W":[],"aC":[],"a6":[]}}'))
0
var y={d:H.az("r*")};(function staticFields(){$.yC=null
$.F7=[$.zG]})()}
$__dart_deferred_initializers__["eYQWuNAANyG5g/dmactXli5kIxQ="] = $__dart_deferred_initializers__.current
//# sourceMappingURL=main.dart.js_1.part.js.map
